docs/how_to/deploy/adreno.rst

```diff
@@ -65,134 +84,667 @@ Reasons of using textures:
 Overall, with textures, it is possible to achieve a significant performance boost
 compared to OpenCL buffer based solutions.
 
-.. _building_tvm_for_adreno:
+.. _about_openclml:
+
+About OpenCLML
+--------------
+
+OpenCLML is a SDK released by Qualcomm that provides accelerated deep learning operators.
+These operators are exposed as an extension "cl_qcom_ml_ops" to standard OpenCL specification.
+Please refer `Accelerate your models with our OpenCL ML SDK <https://developer.qualcomm.com/blog/accelerate-your-models-our-opencl-ml-sdk>`_ for more details.
+
+OpenCLML is integrated into TVM as a `BYOC <https://tvm.apache.org/docs/dev/how_to/relay_bring_your_own_codegen.html?highlight=bring%20your%20own>`_ solution.
+OpenCLML operators can use same context and the operatrors can be enqueued on same command queue if native OpenCL.
+We took advantage of this to avoid any context switching over heads while fallback to native OpenCL.
+
+
+.. _build_deploy:
+
+TVM for Adreno™
```

Comment:
Originally, this document was designed to be a brief introduction to the Adreno usage in TVM. And details related to deployment were described here: https://tvm.apache.org/docs/how_to/deploy_models/deploy_model_on_android.html
One of the benefit of `deploy_model_on_android` document was that user was able to run this script locally with python and run an example on his android. You have removed a link to this document.

Also, this `rst` document starts to be much bigger.  And it is more difficult to get quickly some basic knowledge about Adreno. Previously, section `Build and deploy model for Adreno` briefly showed the peculiarities of the Adreno compilation. For example, differences in the generated kernels with `opencl` and `opencl --device=adreno` and also some information about mixed precision.

My suggestions are:
1. Extent `Building TVM for Adreno` section by the information about building from docker container.
2. For sections about running model on the device and using CLML, create a new doc file or update [deploy_model_on_adreno](https://github.com/apache/tvm/blob/main/gallery/how_to/deploy_models/deploy_model_on_adreno.py)

@srkreddy1238 what do you think about it? I would like to keep this `rst` file as a brief introduction to Adreno and all details move to the `gallery/how_to/deploy_models`.

Reply:
Missed the deploy models part. I will restore it.

Earlier the doc was more texture enhancement centric (too technical). Objective here is to simplify it in a way that a new user will have a good start with TVM and Adreno.

Thanks for the suggestions, it make sense to separate theory and sample code. I will rearrange the same. Feel free to review and advice I want multiple opinions to perfect the docs.

Reply:
> Earlier the doc was more texture enhancement centric (too technical).

It was a try to highlight and explain benefits and technical aspects of Adreno. I'll be happy if after reworking, the document is more helpful and friendly for new users.